src/api/src/attestation/mod.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::Config;
+use anyhow::*;
+use as_types::AttestationResults;
+use async_trait::async_trait;
+use kbs_types::Tee;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+#[cfg(feature = "coco-as")]
+mod coco;
+
+/// Interface for Attestation Services.
+///
+/// Attestation Service implementations should implement that interface.
+#[async_trait]
+pub trait Attest: Send + Sync {
+    /// Verify Attestation Evidence
+    async fn verify(
+        &mut self,
+        tee: Tee,
+        nonce: &str,
+        attestation: &str,
+    ) -> Result<AttestationResults>;
+}
+
+/// Attestation Service
+#[derive(Clone)]
+pub struct AttestationService(pub Arc<Mutex<dyn Attest>>);
+
+impl AttestationService {
+    /// Create and initialize AttestionService
+    pub async fn new(kbs_config: &Config) -> Result<Self> {
+        let attestation_service: Arc<Mutex<dyn Attest>> = {
+            cfg_if::cfg_if! {
+                if #[cfg(any(feature = "coco-as-builtin", feature = "coco-as-builtin-no-verifier"))] {
```

Comment:
An idea hits me: as CoCo-AS and Amber can both support remote connection, we could integration both of the client code and let the configuration to determine the runtime behavior (which one to use) rather than compiling time. In this way an user would not need to worry about the complex different compile features? 

This way the binary would not be much bigger because only client code are included.

How do you think of this?

Reply:
I think this is a very good idea. Should we wait for the Amber PR to be merged and then do that change?

Reply:
Great, agreed!